### PR TITLE
feat: change logic for boost for digg

### PR DESCRIPTION
**Pull request by Darruma**

### CODE REVIEW

File: tests/test_snapshots/test_claims_snapshot.py [hunk 1]
```diff
@@ -32,6 +33,7 @@
 CVX_CRV_PRICE = 12.411460272851376
 SWAPR_WETH_SWAPR_PRICE = 12312.201460272851376
 DIGG_PRICE = 50000
+WBTC_PRICE = 30000
```
Comment (mitche50) [inline tests/test_snapshots/test_claims_snapshot.py:36]: :(

File: tests/test_boost/test_calc_boost.py [hunk 2]
```diff
@@ -62,6 +62,33 @@ def test_allocate_bvecvx_to_users__no_user_boost():
     assert boost_info.get(user) is None
 
 
+def test_allocate_digg_to_users():
```
Comment (mitche50) [inline tests/test_boost/test_calc_boost.py:65]: Let's add a few more tests here:
- Digg amount < native balance
- Digg amount 0, some native balance
- Digg amount == native balance

File: rewards/utils/snapshot_utils.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from decimal import Decimal
+from config.constants.emissions import DIGG_BOOST_VAULTS
+from helpers.enums import Network
+from typing import Dict
+from rewards.snapshot.chain_snapshot import sett_snapshot
+
+
+def digg_snapshot_usd(chain: Network, block: int) -> Dict[str, Decimal]:
```
Comment (SHAKOTN) [inline rewards/utils/snapshot_utils.py:8]: If you return `{}` that's not going to have type of `Dict[str, Decimal]`
Think the type match here will be something like `Union[Dict[str, Decimal], Dict[None, None]`.

File: rewards/classes/Snapshot.py [hunk 4]
```diff
@@ -79,6 +87,14 @@ def convert_to_usd(
...
                 url=discord_url,
             )
+        elif self.token == DIGG:
+            price = Decimal(wbtc_price)
+        elif self.token == BDIGG:
+            _, digg_ppfs = fetch_ppfs()
+            price = Decimal(wbtc_price * digg_ppfs)
+        elif self.token in [BUNI_DIGG_WBTC, BSLP_DIGG_WBTC]:
+            digg_lp_price = prices[self.token]
+            price = 0.5 * digg_lp_price + (digg_lp_price * 0.5 * (wbtc_price / digg_price))
```
Comment (SHAKOTN) [inline rewards/classes/Snapshot.py:97]: I would move `0.5` to constants. Looks like magic number

File: rewards/boost/calc_boost.py [hunk 5]
```diff
@@ -58,6 +69,7 @@ def get_badger_boost_data(stake_ratios: Dict) -> Tuple[Dict, Dict]:
     stake_data_ranges = {}
     for addr, stake_ratio in stake_ratios.items():
         user_stake_range = 0
+        user_boost = 1
```
Comment (btcookies) [inline rewards/boost/calc_boost.py:72]: this is redundant, handled in line 75 and 88

**Review by SHAKOTN — COMMENTED** [review]
Couple comments

File: rewards/boost/calc_boost.py [hunk 6]
```diff
@@ -44,8 +54,9 @@ def calc_stake_ratio(address: str, boost_bals: BoostBalances) -> Decimal:
     native_balance = Decimal(boost_bals.native.get(address, 0))
     non_native_balance = Decimal(boost_bals.non_native.get(address, 0))
     bvecvx_balance = Decimal(boost_bals.bvecvx.get(address, 0))
-    native_balance += calc_bvecvx_native_balance(
-        native_balance, bvecvx_balance)
+    digg_balance = Decimal(boost_bals.digg.get(address, 0))
+    native_balance += calc_bvecvx_native_balance(native_balance, bvecvx_balance)
+    native_balance += calc_digg_native_balance(native_balance, digg_balance)
```
Comment (mitche50) [inline rewards/boost/calc_boost.py:59]: This should be switched.  Per the BIP, `DIGG token contribution to boost will be limited to the total USD value of an addresses BADGER, remBADGER and BADGER NFT` CVX balances should not count towards the limitation

File: rewards/boost/calc_boost.py [hunk 7]
```diff
@@ -165,6 +187,7 @@ def badger_boost(current_block: int, chain: str) -> Dict[str, Any]:
     assign_native_balances_to_users(boost_info, boost_bals.native)
     assign_non_native_balances_to_users(boost_info, boost_bals.non_native)
     allocate_bvecvx_to_users(boost_info, boost_bals.bvecvx)
+    allocate_digg_to_users(boost_info, boost_bals.digg)
```
Comment (mitche50) [inline rewards/boost/calc_boost.py:190]: These should also be switched to conform with the BIP

File: subgraph/queries/harvests.py [hunk 8]
```diff
@@ -34,6 +34,8 @@ def _populate_end_of_previous_harvest(tree_distributions: List[Dict]) -> Dict[st
 
 
 def fetch_tree_distributions(start_timestamp, end_timestamp, chain) -> List[Dict]:
+    print(start_timestamp)
+    print(end_timestamp)
```
Comment (mitche50) [inline subgraph/queries/harvests.py:38]: Stray prints

File: rewards/boost/boost_utils.py [hunk 9]
```diff
@@ -94,8 +101,12 @@ def calc_boost_balances(block: int, chain: str) -> BoostBalances:
         bvecvx = (bvecvx_bals + bvecvx_claimable + bvecvx_lp_bals + fuse_bvecvx)
         bvecvx_usd = bvecvx.convert_to_usd(chain).balances
 
+    digg_claimable_usd = Counter(digg_claimable.convert_to_usd(chain).balances)
+    digg_usd = digg_claimable_usd + Counter(digg_tokens) + Counter(digg_snapshot_usd(chain, block))
+
     native = filter_dust(dict(native), 1)
     non_native = filter_dust(dict(non_native), 1)
     bvecvx_usd = filter_dust(dict(bvecvx_usd), 1)
+    digg_usd = filter_dust(dict(bvecvx_usd), 1)
```
Comment (mitche50) [inline rewards/boost/boost_utils.py:110]: This should be `digg_usd`

File: rewards/classes/Snapshot.py [hunk 10]
```diff
@@ -79,6 +87,15 @@ def convert_to_usd(
...
+        elif self.token == DIGG:
+            price = Decimal(wbtc_price)
+        elif self.token == BDIGG:
+            _, digg_ppfs = fetch_ppfs()
+            price = Decimal(wbtc_price * digg_ppfs)
+        elif self.token in [BUNI_DIGG_WBTC, BSLP_DIGG_WBTC]:
+            digg_lp_price = prices[self.token]
+            price = DIGG_LP_PRICE_RATIO * digg_lp_price \
+                + (digg_lp_price * DIGG_LP_PRICE_RATIO * (wbtc_price / digg_price))
         else:
```
Comment (mitche50) [inline rewards/classes/Snapshot.py:99]: What's the `+` for here

File: rewards/classes/Snapshot.py [hunk 11]
```diff
@@ -79,6 +87,15 @@ def convert_to_usd(
...
                 url=discord_url,
             )
+        elif self.token == DIGG:
+            price = Decimal(wbtc_price)
+        elif self.token == BDIGG:
+            _, digg_ppfs = fetch_ppfs()
+            price = Decimal(wbtc_price * digg_ppfs)
+        elif self.token in [BUNI_DIGG_WBTC, BSLP_DIGG_WBTC]:
+            digg_lp_price = prices[self.token]
+            price = DIGG_LP_PRICE_RATIO * digg_lp_price \
```
Comment (SHAKOTN) [inline rewards/classes/Snapshot.py:97]: Maybe we need to explicitly cast this expr to `Decimal` as well?

**Review by mitche50 — APPROVED** [review]
Changes all LGTM

**Review by SHAKOTN — APPROVED** [review]

**Pull request merged**